src/components/ChatDetailsQuickActionsBar.tsx

```diff
@@ -0,0 +1,64 @@
+import React, {useState} from 'react';
+import {View} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as Report from '@userActions/Report';
+import type {Report as OnyxReportType} from '@src/types/onyx';
+import Button from './Button';
+import ConfirmModal from './ConfirmModal';
+import * as Expensicons from './Icon/Expensicons';
+
+type ChatDetailsQuickActionsBarProps = {
+    report: OnyxReportType;
+};
+
+function ChatDetailsQuickActionsBar({report}: ChatDetailsQuickActionsBarProps) {
+    const styles = useThemeStyles();
+    const [isLastMemberLeavingGroupModalVisible, setIsLastMemberLeavingGroupModalVisible] = useState(false);
+    const {translate} = useLocalize();
+    const isPinned = !!report.isPinned;
+    return (
+        <View style={[styles.flexRow, styles.ph5, styles.mb5]}>
+            <View style={[styles.flex1, styles.pr3]}>
+                <ConfirmModal
+                    danger
+                    title={translate('groupChat.lastMemberTitle')}
+                    isVisible={isLastMemberLeavingGroupModalVisible}
+                    onConfirm={() => {
+                        setIsLastMemberLeavingGroupModalVisible(false);
+                        Report.leaveGroupChat(report.reportID);
+                    }}
+                    onCancel={() => setIsLastMemberLeavingGroupModalVisible(false)}
+                    prompt={translate('groupChat.lastMemberWarning')}
+                    confirmText={translate('common.leave')}
+                    cancelText={translate('common.cancel')}
+                />
+                <Button
+                    onPress={() => {
+                        if (Object.keys(report?.participants ?? {}).length === 1) {
+                            setIsLastMemberLeavingGroupModalVisible(true);
+                            return;
+                        }
+
+                        Report.leaveGroupChat(report.reportID);
```

Comment:
When we leave a group chat we replace the current screen with the most recent report so the group chat is no longer in stack. However in this case the function is called from a RHP screen. 

If the stack is as follow:
- `report 1 -> report 1 details page`

and we leave the group, the stack will be:
- `report 1 -> report 2`

report 1 can still be accessed by going back (bug).

To fix this we opted to dismiss any RHP modal before proceeding